File: AtCoder/AtCoder_Improvement_Guidelines/2-3/D - パ研軍旗.py
s=[*zip(*[*open(0)][1:])][:-1]
dp=[0]*3; r=0,1,2
for _ in s:
    dp=[min(dp[j] for j in r if j!=i)+5-_.count("RBW"[i]) for i in r]
print(min(dp))

File: Archive/AtCoder/AtCoder_Improvement_Guidelines/2-3/D - パ研軍旗.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
  int N; cin>>N;
  string S[5]; for(string &Si:S)cin>>Si;
  int d[3]{};
  for(int i=0;i<N;i++){
    int c[3]{},td[3]{};
    for(string Sj:S)for(int k:{0,1,2})if(Sj[i]!="RBW"[k])c[k]++;
    for(int j:{0,1,2}){
      int m=5e3; for(int k:{0,1,2})if(k!=j)m=min(m,d[k]);
      td[j]=m+c[j];
    }
    memcpy(d,td,sizeof(td));
  }
  cout<<min({d[0],d[1],d[2]})<<endl;
}